kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -274,7 +279,7 @@ public MockResponse handleWatch(String path) {
     }
     WatchEventsListener watchEventListener = new WatchEventsListener(context, query, watchEventListeners, LOGGER,
       watch -> {
-        synchronized (KubernetesCrudDispatcher.this) {
+        synchronized (map) {
```

Comment:
I wasn't sure of the ramifications of this change, as in, I'm unclear why there was a synchronized block previously the way it was, but I believe it should only be synchronized around the map. I didn't want to use double synchronization.